client/src/components/molecules/DonationMatchingRequestsTable.tsx

```diff
@@ -0,0 +1,168 @@
+import { TextField } from "../atoms/TextField";
+
+interface Props {
+    requests: Request[];
+    donationForm?: DonationForm;
+    isErroneous: boolean; // indicates whether the selected quantities are erroneous
+    onQuantitySelected: (newQuantity: number, requestId: string) => void;
+}
+
+type RequestQuantities = {
+    quantity: number; // current value selected in dropdown
+    totalRequired: number; // total unmatched requests
+};
+
+const DonationMatchingRequestsTable: FunctionComponent<Props> = (props: Props) => {
+    const headingList = ["Client Name", "Quantity", "Total Required", "Date Requested"];
+
+    const [requests, setRequests] = useState<Request[]>([]);
+    const [reqCurrentlyEditing, setReqCurrentlyEditing] = useState<string>("");
+
+    useEffect(() => {
+        // sort requests by most recent date
+        const sortedRequests: Request[] = props.requests
+            .filter((request) => !request.deleted && !request.fulfilled)
+            .sort((a, b) => {
+                return b!.createdAt!.valueOf() - a!.createdAt!.valueOf();
+            });
+        setRequests(sortedRequests);
+    }, []);
+
+    /**
+     * Calculates the numerical values to display in the table row for a specific request
+     */
+    const valuesToDisplay = (request: Request): RequestQuantities => {
+        const { matchedDonations } = request;
+
+        const totalQuantityMatched = matchedDonations?.reduce((total, contribution) => {
+            return total + contribution.quantity;
+        }, 0) as number;
+
+        const requestTotal = request?.quantity ?? 0;
+        const totalRequired = requestTotal - totalQuantityMatched;
+
+        if (props.donationForm) {
+            // calculate the number of items assigned to the current item being matched
+            const quantityAssignedToForm = matchedDonations?.reduce((total, contribution) => {
+                return contribution.donationForm == props?.donationForm?._id ? total + contribution.quantity : total;
+            }, 0);
+
+            return {
+                quantity: quantityAssignedToForm as number,
+                totalRequired
+            };
+        }
+        return {
+            quantity: totalQuantityMatched,
+            totalRequired
+        };
+    };
+
+    const isQuantityErroneous = (values: RequestQuantities): boolean => {
+        const { quantity } = values;
+        return quantity > 0 && props.isErroneous;
+    };
+
+    const maxDropdownQuantity = (values: RequestQuantities): number => {
+        const { quantity, totalRequired } = values;
+        return quantity + totalRequired;
+    };
+
+    const onQuantityChange = (quantity: number, requestId: string) => {
+        props.onQuantitySelected(quantity, requestId);
+        setReqCurrentlyEditing("");
+    };
+
+    return (
+        <div className="donation-matching-request-list">
+            <Table responsive className="donation-matching-request-table">
```

Comment:
Might want to wrap a Dropdown component if requestType is not null. See RequestTypeDropdown.tsx because it is pretty similar to that.

Reply:
I'm using `DonationMatchingRequestsTable.tsx` similar to `RequestsTable.tsx`, where the component simply displays a list of requests (whatever is passed in as props). To display the dropdown, I'll build a component called `MatchingRequestTypeDropdown.tsx` that behaves pretty much like `RequestTypeDropdown.tsx`, except it uses this table instead of `RequestsTable` (and without editing/deletion, since matching page doesn't require that)

Reply:
Sounds good, I'll resolve this comment